### ISSUE TITLE
Imbalance doesn't work with categorical data not encoded as categorical arrays 

### ISSUE THREAD

**Issue by sylvaticus** [JuliaAI/Imbalance.jl#98]
Take this example:

```
a = DataFrame(a =["a","b","c"], b=[1,2,3])
b = ["a","a","b"]
Xover, yover = random_oversample(a, b)   
```

This fails because ` ScientificTypes.schema(X).scitypes` fails, but the algorithms emploied doesn't really care about scientific types.

**Comment by EssamWisam** [comment]
For consistency, as in the documentation of each method, all methods at most assume that
> X: A matrix of real numbers or a table with element [scitypes](https://juliaai.github.io/ScientificTypes.jl/) that subtype Union{Finite, Infinite}.

Here the column `a` has scientific type `Textual` and should be coerced to `MultiClass` first. I see that the error does not seem to signal that directly and will consider making a PR that throws a better one.

We can also add support for the `Textual` type but it may not be straightforward given the current implementation.

**Comment by ablaom** [comment]
Yeah, you that one algorithm doesn't need much, but for consistency with the other algorithms, we require `y` to be a `CategoricalVector`.

**Comment by ablaom** [comment]
So, for example, this works:

```julia
using CategoricalArrays
a = DataFrame(a =categorical(["a","b","c"]), b=[1,2,3]);
```

**Comment by ablaom** [comment]
Or:

```julia
using ScientificTypes
coerce!(a, a: => Multiclass)
```

**Comment by ablaom** [comment]
But I agree with @sylvaticus  that it's worth special casing this algorithm which can deal with arbitrary column types.
